src/app/modules/convai-checker/perspectiveapi.service.ts

```diff
@@ -130,11 +129,26 @@ export class PerspectiveApiService {
         console.error('No server url specified for a non-direct API call.'
                       + ' Defaulting to current hosted address');
       }
-      const headers = new HttpHeaders();
-      headers.append('Content-Type', 'application/json');
 
-      return this.httpClient.post(
-        serverUrl + '/suggest_score', data, {headers});
+      return this.sendRequest(
+          data, `${serverUrl}/suggest_score`, 'suggestScore');
     }
   }
+
+  private sendRequest<T, R>(data: T, endpoint: string, action: string):
+    Observable<R> {
+    return this.recaptchaV3Service === undefined ||
```

Comment:
Could it definitely be either undefined or null, or would it just be undefined? If it could be both then == null would make this briefer (it means === null or === undefined).

Reply:
I did a little test and it would be null. Changed the check accordingly.